<launch>

<arg name="input_dim"/>
<arg name="max_evals"/>

<arg name="num_init_samples" default="30"/>
<arg name="verbose" default="false"/>
<arg name="output_path" default="out.pkl"/>
<arg name="progress_path" default="prog.pkl"/>
<arg name="resume" default="true"/>
<arg name="critic_service" default="evaluator/get_critique"/>
<arg name="beta" default="1.0"/>

<arg name="enable_context" default="false"/>
<arg name="context_stream" default="void"/>

<arg name="enable_refine" default="false"/>
<arg name="context_mode" default="ignore"/>

<arg name="run_on_start" default="true"/>

<node pkg="optim" type="BayesianOptimizer.py" name="optimizer" output="screen" required="true">
    <param name="output_path" value="$(arg output_path)"/>
    <param name="load_path" value="$(arg load_path)" if="$(arg resume)"/>
    <param name="progress_path" value="$(arg progress_path)"/>

    <param name="convergence/max_evaluations" value="$(arg max_evals)"/>
    <param name="dim" value="$(arg input_dim)"/>
    <param name="exploration_rate" value="$(arg beta)"/>
    <param name="initialization/num_samples" value="$(arg num_init_samples)"/>

    <param name="interface/topic" value="$(arg critic_service)"/>
    <param name="interface/verbose" value="$(arg verbose)"/>

    <param name="reward_model/verbose" value="$(arg verbose)"/>
    <param name="reward_model/enable_refine" value="$(arg enable_refine)"/>

    <param name="run_on_start" value="$(arg run_on_start)"/>
    <param name="context_mode" value="$(arg context_mode)"/>
    <param name="input_stream" value="$(arg context_stream)" if="$(arg enable_context)"/>
    <rosparam>
        interface:
            n_retries: 10

        init_beta: 1.0
        desired_beta: 3.0
        desired_beta_rounds: 100

        lower_bounds: -1
        upper_bounds: 1
        sequential_mode: false

        context_normalization:
            mode: minmax

        convergence:
            x_tol: 0

        reward_model:
            type: gaussian_process
            batch_retries: 9
            refine_ll_delta: -inf
            refine_retries: 3

        auxiliary_optimizer:
            type: bfgs
            mode: max
            num_restarts: 10
            ftol: 1e-6
            gtol: 1e-6
            maxiter: 500

        initialization:
            method: uniform
    </rosparam>
</node>

</launch>